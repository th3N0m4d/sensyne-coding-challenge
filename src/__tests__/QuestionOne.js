import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Button from '@material-ui/core/Button'

import QuestionOne from '../pages/questionOne/questionOne'

const expectToMatchSnapshot = component => {
  expect(renderer.create(component).toJSON()).toMatchSnapshot()
}

describe('QuestionOne', () => {
  it('should render component', () => {
    expectToMatchSnapshot(<QuestionOne />)
  })

  it('should trigger onClick', () => {
    const onClickSpy = jest.fn()
    const wrapper = shallow(<QuestionOne onClick={onClickSpy} />)

    const button = wrapper.find(Button)

    button.simulate('click')

    expect(onClickSpy).toHaveBeenCalled()
  })
})
