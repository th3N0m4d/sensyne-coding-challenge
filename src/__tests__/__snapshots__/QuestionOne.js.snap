// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionOne should render component 1`] = `
<div
  className="sc-AxjAm jTNVWA"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      I've been clicked: undefined times
    </span>
  </button>
</div>
`;

exports[`QuestionOne should render component with initial count of 7 1`] = `
<div
  className="sc-AxjAm jTNVWA"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      I've been clicked: 7 times
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
